VideoLocker/src/main/java/org/edx/mobile/view/custom/URLInterceptorWebViewClient.java

```diff
@@ -234,7 +234,11 @@ private boolean isExternalLink(String strUrl) {
     private boolean isEnrollLink(String strUrl) {
         if (actionListener != null && strUrl != null &&
                 strUrl.startsWith(URL_TYPE_ENROLL)) {
-            Uri uri = Uri.parse(strUrl);
+
+            // The edx.org course catalog does not encode plus signs in the course_id parameter
+            // We encode them here before parsing the URI, so they aren't converted to space characters
+            // TODO: Remove this hack once it's fixed on the server: https://openedx.atlassian.net/browse/MA-1901
+            Uri uri = Uri.parse(strUrl.replace("+", "%2B"));
```

Comment:
> Don't you need to convert the %2Bs back to +s after extracting the courseId

`getQueryParameter` will decode the parameters, converting this back to a `+` (which is precisely why this broke, it's currently decoding the `+` to a space). And yes, the `courseId` variable here is used in the POST.

I am able to enroll courses with this change applied.